lib/schema/template.go

```diff
@@ -176,8 +177,13 @@ func valueFromFile(path, basePath string) (data []byte, mimeType string, err err
 
 // valueFromHTTP returns the content downloaded from the provided URL
 // and its MIME type
-func valueFromHTTP(url string) (data []byte, mimeType string, err error) {
-	response, err := http.Get(url)
+func valueFromHTTP(ctx context.Context, url string) (data []byte, mimeType string, err error) {
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+	req = req.WithContext(ctx)
+	response, err := (&http.Client{}).Do(req)
```

Comment:
Instead of creating an empty client it might be preferable to use http.DefaultClient? Alternatively might want to initialize the client with a Timeout, although the context should largely cover that, but might help if someone just passes a TODO context.

Reply:
`http.DefaultClient` is also the empty `http.Client` value - e.g. it has the same defaults. I'll update to use `lib/httplib` for timeouts though.